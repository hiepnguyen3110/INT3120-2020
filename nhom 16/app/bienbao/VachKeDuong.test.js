import React from 'react';
import renderer from 'react-test-renderer';

import VachKeDuong from './VachKeDuong';

it('renders correctly', () => {
    const tree = renderer.create(<VachKeDuong />).toJSON();
    expect(tree).toMatchSnapshot();
  });