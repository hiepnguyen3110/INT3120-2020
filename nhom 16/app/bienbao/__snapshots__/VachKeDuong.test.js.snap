// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "Img": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/12-vach-ke-duong-12.jpg",
          "Key": "a1",
          "Name": "Vạch kẻ đường 1.2",
          "Noidung": "Vạch liền, màu trắng, rộng 20 cm, dùng để xác định mép phần xe chạy trên các trục đường. Xe chạy được phép cắt ngang hoặc đè lên vạch khi cần thiết..",
        },
        Object {
          "Img": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/14-vach-ke-duong-14.jpg",
          "Key": "a2",
          "Name": "Vạch kẻ đường 1.4",
          "Noidung": "Là vạch liên tục màu vàng có chiều rộng 10 cm, để xác định nơi cấm dừng và cấm đỗ xe.",
        },
        Object {
          "Img": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/15-vach-ke-duong-15.jpg",
          "Key": "a3",
          "Name": "Vạch kẻ đường 1.5",
          "Noidung": "Là vạch đứt quãng, màu trắng, rộng 10 cm, tỷ lệ L1:L2 = 1:3. Vạch dùng để phân chia 2 dòng phương tiện giao thông từ 2 hướng ngược chiều nhau trên các đường có 2 hoặc 3 làn xe chạy. Xác định ranh giới làn xe khi có 2 hoặc trên 2 làn xe chạy theo một hướng.",
        },
        Object {
          "Img": "https://i.xeoto.com.vn/bien-bao-giao-thong/bien-bao-phu/503a-huong-tac-dung-cua-bien.jpg",
          "Key": "a4",
          "Name": "Vạch kẻ đường 1.6",
          "Noidung": "Là vạch đứt quãng màu trắng, rộng 10 cm. Tỷ lệ L1:L2 = 3:1, dùng để báo hiệu gần đến vạch 1-1 hay 1-11, để phân chia dòng xe ngược chiều hay cùng chiều.",
        },
        Object {
          "Img": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/17-vach-ke-duong-17.jpg",
          "Key": "a5",
          "Name": "vạch kẻ đường 1.7",
          "Noidung": "Là vạch đứt quãng màu trắng rộng 0,1m, khoảng cách giữa hai vạch là 0,5m. Vạch được kẻ Theo đường cong Theo chiều xe chạy ở chỗ giao nhau khi lái xe cần định hướng chung để đảm bảo an toàn khi qua chỗ giao nhau.",
        },
        Object {
          "Img": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/13-vach-ke-duong-13.jpg",
          "Key": "a6",
          "Name": "Vạch kẻ đường 1.3",
          "Noidung": "Là vạch kép (2 vạch liên tục) màu trắng, có chiều rộng bằng nhau và bằng 10 cm, cách nhau là 10 cm, dùng để phân chia 2 dòng phương tiện giao thông từ 2 hướng ngược chiều nhau trên những đường có từ 4 làn đường trở lên. Xe chạy không được đè qua vạch.",
        },
        Object {
          "Img": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/18-vach-ke-duong-18.jpg",
          "Key": "a7",
          "Name": "vạch kẻ đường 1.8",
          "Noidung": "Là vạch đứt quãng màu trắng rộng0,4m. Vạch dùng để quay định danh giới làn xe tăng tốc độ hoặc giảm tốc độ (gọi là chuyển tới làn đường) và làn xe chính của phần xe chạy.",
        },
        Object {
          "Img": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/18-vach-ke-duong-18.jpg",
          "Key": "a8",
          "Name": "vạch kẻ đường 1.9",
          "Noidung": "Là loại vạch kép (hai vạch) đứt quãng, song song, màu trắng rộng 0,1m và cách nhau 0,1 m.Vạch quay định danh giới làn xe dự trữ mà trên làn này chiều xe chạy có thể thay đổi hoặc hoặc chiều thuận hoặc chiều đi ngược lại. Sự thay đổi hướng xe được điều khiển bằng tín hiệu đèn xanh và đỏ đặt trên làn xe.",
        },
        Object {
          "Img": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/110-vach-ke-duong-110.jpg",
          "Key": "a9",
          "Name": "vạch kẻ đường 1.10",
          "Noidung": "Là vạch đứt quãng màu vàng. Vạch xác định vị trí hay khu vực cấm đỗ xe.",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 80,
                  "marginTop": 5,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/12-vach-ke-duong-12.jpg",
                  }
                }
                style={
                  Object {
                    "height": "91%",
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2.7,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                Vạch kẻ đường 1.2
              </Text>
              <Text
                style={
                  Object {
                    "color": "#A4A4A4",
                    "fontSize": 17,
                  }
                }
              >
                Vạch liền, màu trắng, rộng 20 cm, dùng để xác định mép phần xe chạy trên các trục đường. Xe chạy được phép cắt ngang hoặc đè lên vạch khi cần thiết..
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A4A4A4",
                "height": 2,
                "marginTop": 10,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 80,
                  "marginTop": 5,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/14-vach-ke-duong-14.jpg",
                  }
                }
                style={
                  Object {
                    "height": "91%",
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2.7,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                Vạch kẻ đường 1.4
              </Text>
              <Text
                style={
                  Object {
                    "color": "#A4A4A4",
                    "fontSize": 17,
                  }
                }
              >
                Là vạch liên tục màu vàng có chiều rộng 10 cm, để xác định nơi cấm dừng và cấm đỗ xe.
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A4A4A4",
                "height": 2,
                "marginTop": 10,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 80,
                  "marginTop": 5,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/15-vach-ke-duong-15.jpg",
                  }
                }
                style={
                  Object {
                    "height": "91%",
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2.7,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                Vạch kẻ đường 1.5
              </Text>
              <Text
                style={
                  Object {
                    "color": "#A4A4A4",
                    "fontSize": 17,
                  }
                }
              >
                Là vạch đứt quãng, màu trắng, rộng 10 cm, tỷ lệ L1:L2 = 1:3. Vạch dùng để phân chia 2 dòng phương tiện giao thông từ 2 hướng ngược chiều nhau trên các đường có 2 hoặc 3 làn xe chạy. Xác định ranh giới làn xe khi có 2 hoặc trên 2 làn xe chạy theo một hướng.
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A4A4A4",
                "height": 2,
                "marginTop": 10,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 80,
                  "marginTop": 5,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.xeoto.com.vn/bien-bao-giao-thong/bien-bao-phu/503a-huong-tac-dung-cua-bien.jpg",
                  }
                }
                style={
                  Object {
                    "height": "91%",
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2.7,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                Vạch kẻ đường 1.6
              </Text>
              <Text
                style={
                  Object {
                    "color": "#A4A4A4",
                    "fontSize": 17,
                  }
                }
              >
                Là vạch đứt quãng màu trắng, rộng 10 cm. Tỷ lệ L1:L2 = 3:1, dùng để báo hiệu gần đến vạch 1-1 hay 1-11, để phân chia dòng xe ngược chiều hay cùng chiều.
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A4A4A4",
                "height": 2,
                "marginTop": 10,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 80,
                  "marginTop": 5,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/17-vach-ke-duong-17.jpg",
                  }
                }
                style={
                  Object {
                    "height": "91%",
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2.7,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                vạch kẻ đường 1.7
              </Text>
              <Text
                style={
                  Object {
                    "color": "#A4A4A4",
                    "fontSize": 17,
                  }
                }
              >
                Là vạch đứt quãng màu trắng rộng 0,1m, khoảng cách giữa hai vạch là 0,5m. Vạch được kẻ Theo đường cong Theo chiều xe chạy ở chỗ giao nhau khi lái xe cần định hướng chung để đảm bảo an toàn khi qua chỗ giao nhau.
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A4A4A4",
                "height": 2,
                "marginTop": 10,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 80,
                  "marginTop": 5,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/13-vach-ke-duong-13.jpg",
                  }
                }
                style={
                  Object {
                    "height": "91%",
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2.7,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                Vạch kẻ đường 1.3
              </Text>
              <Text
                style={
                  Object {
                    "color": "#A4A4A4",
                    "fontSize": 17,
                  }
                }
              >
                Là vạch kép (2 vạch liên tục) màu trắng, có chiều rộng bằng nhau và bằng 10 cm, cách nhau là 10 cm, dùng để phân chia 2 dòng phương tiện giao thông từ 2 hướng ngược chiều nhau trên những đường có từ 4 làn đường trở lên. Xe chạy không được đè qua vạch.
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A4A4A4",
                "height": 2,
                "marginTop": 10,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 80,
                  "marginTop": 5,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/18-vach-ke-duong-18.jpg",
                  }
                }
                style={
                  Object {
                    "height": "91%",
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2.7,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                vạch kẻ đường 1.8
              </Text>
              <Text
                style={
                  Object {
                    "color": "#A4A4A4",
                    "fontSize": 17,
                  }
                }
              >
                Là vạch đứt quãng màu trắng rộng0,4m. Vạch dùng để quay định danh giới làn xe tăng tốc độ hoặc giảm tốc độ (gọi là chuyển tới làn đường) và làn xe chính của phần xe chạy.
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A4A4A4",
                "height": 2,
                "marginTop": 10,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 80,
                  "marginTop": 5,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/18-vach-ke-duong-18.jpg",
                  }
                }
                style={
                  Object {
                    "height": "91%",
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2.7,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                vạch kẻ đường 1.9
              </Text>
              <Text
                style={
                  Object {
                    "color": "#A4A4A4",
                    "fontSize": 17,
                  }
                }
              >
                Là loại vạch kép (hai vạch) đứt quãng, song song, màu trắng rộng 0,1m và cách nhau 0,1 m.Vạch quay định danh giới làn xe dự trữ mà trên làn này chiều xe chạy có thể thay đổi hoặc hoặc chiều thuận hoặc chiều đi ngược lại. Sự thay đổi hướng xe được điều khiển bằng tín hiệu đèn xanh và đỏ đặt trên làn xe.
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A4A4A4",
                "height": 2,
                "marginTop": 10,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 80,
                  "marginTop": 5,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://i.xeoto.com.vn/bien-bao-giao-thong/vach-ke-duong/110-vach-ke-duong-110.jpg",
                  }
                }
                style={
                  Object {
                    "height": "91%",
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2.7,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                vạch kẻ đường 1.10
              </Text>
              <Text
                style={
                  Object {
                    "color": "#A4A4A4",
                    "fontSize": 17,
                  }
                }
              >
                Là vạch đứt quãng màu vàng. Vạch xác định vị trí hay khu vực cấm đỗ xe.
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#A4A4A4",
                "height": 2,
                "marginTop": 10,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
